Sources/SPTDataLoaderServiceSessionSelector.m

```diff
@@ -56,6 +71,13 @@ - (instancetype)initWithConfiguration:(NSURLSessionConfiguration *)configuration
 
 - (NSURLSession *)URLSessionForRequest:(SPTDataLoaderRequest *)request
 {
+    if (request.backgroundPolicy == SPTDataLoaderRequestBackgroundPolicyAlways) {
+        NSURLSession *backgroundSession = self.backgroundSession;
```

Comment:
it can be assumed it background session is always there if it is created internally instead of being injected at `init`

Reply:
In this setup, the background session will only be created if a configuration object for it is provided at init. It looks safer to me to provide a new initializer instead of changing this logic for everyone who's already using data loader.

Reply:
Not really. It opens up to bugs and clutter the API.
I actually I think the missing of an internal background session is kind of a bug if background tasks are allowed